internal/mailing/emailParser.go

```diff
@@ -19,24 +19,20 @@ func path(filename string) string {
 	return filepath.Join(basePath, "emails", filename)
 }
 
-type HTMLTemplateParser interface {
-	Parse(templateName string, data any) ([]byte, error)
-}
-
 type htmlTemplateParser struct {
 	l config.Logger
 }
 
-func NewHTMLTemplateParser(l config.Logger) HTMLTemplateParser {
+func NewHTMLTemplateParser(l config.Logger) *htmlTemplateParser {
 	return &htmlTemplateParser{l: l}
 }
 func (p htmlTemplateParser) Parse(templateName string, data any) ([]byte, error) {
 	var buffer bytes.Buffer
-	filename := path(templateName + ".html")
+	templateFile := path(templateName + ".html")
 	if err := template.
-		Must(template.ParseFiles(filename)).
+		Must(template.ParseFiles(templateFile)).
```

Comment:
Must is always panics on error. Are you sure you want to use it here?
Also seems that htmlTemplateParser is no more than just wrapper around parsing html. You sure you need this as separate component, not just private method where you are using this?